### ISSUE TITLE
most IBM dynamic tests fail with Sessions prototype

### ISSUE THREAD

**Issue by hppritcha** [hpc/ompi#28]
These tests in the dynamic section of IBM fail with the Sessions prototype:
```
FAIL: spawn
FAIL: spawn_with_env_vars
FAIL: spawn_multiple
FAIL: no-disconnect
FAIL: intercomm_create
FAIL: spawn_argvnull_mpifh
FAIL: spawn_argvsnull_mpifh
FAIL: spawn_argvnull_usempi
FAIL: spawn_argvsnull_usempi
```

**Comment by hppritcha** [comment]
these still fail after today's rebase of sessions_new

**Comment by hppritcha** [comment]
turns out this is uncovering an issue with the way the prototype is handling common finalize.  After some refactoring of the initialization process (mpi3) to make sure that spawned processes don't blow up when trying to set up MPI_COMM_PARENT and handling the intercommunicator setup, found out that the TCP BTL was exploding at finalize.
The problem there was that currently the shutdown process in the prototype invokes ```ompi_proc_finalize``` prior to cleaning up the opal layer.  So some proc_opal's being used inside the TCP BTL were already freed up by the preceding call to ```ompi_proc_finalize```.  In master, no sessions, ompi_proc_finalize is one of the last calls in the mpi finalization procedure, after the opal/btl cleanup.  delaying the call to ``ompi_proc_finalize```  to after the opal/btl finalization fixes the problem.

**Comment by hppritcha** [comment]
fixed via ad1224f
